### ISSUE TITLE
Merge and Override Template properties from the instance of the template properties

### ISSUE THREAD

**Issue by dogezen** [Kiamo2/YATI#18]
**Is your feature request related to a problem? Please describe.**

In Tiled you can define template objects, where each template object may have zero or more properties.

When you make an instance of an object from a template, Tiled makes the template properties available in the object instance so that the property values can stay as they are defined in the template or be overridden in the instanced version. Moreover, you can define additional properties that are not part of the template.

In YATI, at the moment, when we instance a template we have the `obj` that uses the template, the template is loaded and then an instance of the object is created in the godot layer from the template. However, any property overrides / additions defined in obj are not carried over into the template, and hence are not available.

**Describe the solution you'd like**

Merge and override any property changes from the `obj` into the `template_obj` before instancing.

**Working solution**

```python
# add the following code just before handling the creation of the object from template
# this code is approximately lines 663 in TilemapCreator.gd

				# override and merge properties defined in obj with properties defined in template
				# since obj may override and define additional properties to those defined in template
				if obj.has("properties"):
					if not template_obj.has("properties"):
						# template comes without properties, since obj has properties
						# then merge them into the template
						template_obj["properties"] = obj.properties
					else:
						# merge obj properties that may have been overridden in the obj instance
						# and add any additional properties defined in instanced obj that are 
						# not defined in template
						for prop in obj["properties"]:
							var found = false
							for templ_prop in template_obj["properties"]:
								if prop.name == templ_prop.name:
									templ_prop.value = prop.value
									found = true
									break
							if not found:
								template_obj["properties"].append(prop)
				handle_object(template_obj, layer_node, template_tileset, Vector2(obj_x, obj_y))
```

**Comment by Kiamo2** [comment]
I have to admit that all my test cases up til now have neglected extensive template examples.
So I'm very thankful for your valuable input.

Would you think that this is the last problem in connection with templates or are there more to come?

**Comment by dogezen** [comment]
I hope this is the last of it with templates? I have only tested object templates, I have not tried having templates with the `template_dict.has("tilesets"):` stuff, so maybe there might be some more work to do there?

I am happy to contribute more if I find things as they come up, let me know if you'd prefer that (ie. submitting some PRs)

**Comment by Kiamo2** [comment]
Issue is now fixed in v1.5.3.
Again thanks for the solution!
For the moment I'd like to get along without PRs as I'm afraid otherwise to lose the overview.
